jetstream/consumer_config.go

```diff
@@ -20,85 +20,243 @@ import (
+		MaxDeliver int `json:"max_deliver,omitempty"`
+
+		// BackOff specifies the optional back-off intervals for retrying
+		// message delivery after a failed acknowledgement. It overrides
+		// AckWait.
+		//
+		// The number of intervals specified must be lower or equal to
+		// MaxDeliver. If the number of intervals is lower, the last interval is
+		// used for all remaining attempts.
+		BackOff []time.Duration `json:"backoff,omitempty"`
+
+		// FilterSubject can be used to filter messages delivered form the
+		// stream. It has to overlap with the subjects defined on the stream.
+		// FilterSubject is exclusive with FilterSubjects.
+		FilterSubject string `json:"filter_subject,omitempty"`
+
+		// ReplayPolicy defines the rate at which messages are sent to the
+		// consumer. If ReplayOriginalPolicy is set, messages are sent at the
+		// same rate as they were stored in the stream. If ReplayInstantPolicy
+		// is set, messages are sent as fast as possible. Defaults to
+		// ReplayInstantPolicy.
+		ReplayPolicy ReplayPolicy `json:"replay_policy"`
+
+		// RateLimit specifies an optional maximum rate of message delivery in
+		// bits per second.
+		RateLimit uint64 `json:"rate_limit_bps,omitempty"`
+
+		// SampleFrequency is an optional frequency for sampling how often
+		// acknowledgements are sampled for observability. See
+		// https://docs.nats.io/running-a-nats-service/nats_admin/monitoring/monitoring_jetstream
+		SampleFrequency string `json:"sample_freq,omitempty"`
+
+		// MaxWaiting is a maximum number of pull requests waiting to be
+		// fulfilled. Defaults to 512.
+		MaxWaiting int `json:"max_waiting,omitempty"`
+
+		// MaxAckPending is a maximum number of outstanding unacknowledged
+		// messages. Once this limit is reached, the server will suspend sending
+		// messages to the consumer. Defaults to 1000. Set to -1 for unlimited.
+		MaxAckPending int `json:"max_ack_pending,omitempty"`
+
+		// HeadersOnly indicates whether only headers of messages should be sent
+		// (and no payload). Defaults to false.
+		HeadersOnly bool `json:"headers_only,omitempty"`
+
+		// MaxRequestBatch is the optional maximum batch size a single pull
+		// request can make. When set with MaxRequestMaxBytes, the batch size
+		// will be constrained by whichever limit is hit first.
+		MaxRequestBatch int `json:"max_batch,omitempty"`
+
+		// MaxRequestExpires is the maximum duration a single pull request will
+		// wait for messages to be available to pull.
+		MaxRequestExpires time.Duration `json:"max_expires,omitempty"`
+
+		// MaxRequestMaxBytes is the optional maximum total bytes that can be
+		// requested in a given batch. When set with MaxRequestBatch, the batch
+		// size will be constrained by whichever limit is hit first.
+		MaxRequestMaxBytes int `json:"max_bytes,omitempty"`
+
+		// InactiveThreshold is a duration which instructs the server to clean
+		// up the consumer if it has been inactive for the specified duration.
```

Comment:
Changed to:
```go

		// InactiveThreshold is a duration which instructs the server to clean
		// up the consumer if it has been inactive for the specified duration.
		// Durable consumers will not be cleaned up by default, but if
		// InactiveThreshold is set, they will be. If not set, this will inherit
		// settings from stream's ConsumerLimits. If neither are set, server
		// default is 5 seconds.
		//
		// A consumer is considered inactive there are not pull requests
		// received by the server (for pull consumers), or no interest detected
		// on deliver subject (for push consumers), not if there are no
		// messages to be delivered.
```